.github/scripts/compile_all_variants.sh

```
@@ -0,0 +1,89 @@
+fi
+
+summary_file="${GITHUB_STEP_SUMMARY:-/dev/stdout}"
+{
+  echo "## eloqkv from-scratch compile matrix (ubuntu:24.04, prefix: \`${PREFIX}\`)"
+  echo ""
+  echo "| Variant | WITH_DATA_STORE | WITH_LOG_STATE | Result | Duration |"
+  echo "|---|---|---|---|---|"
+} >> "${summary_file}"
+
+overall_status=0
+for entry in "${VARIANTS[@]}"; do
```

Shellcheck (0.11.0)

[info] The mentioned syntax error was in this for loop.

(SC1009)